pkg/apis/core/validation/validation.go

```diff
@@ -7479,6 +7486,10 @@ func ValidatePodLogOptions(opts *core.PodLogOptions) field.ErrorList {
 		if *opts.SinceSeconds < 1 {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("sinceSeconds"), *opts.SinceSeconds, "must be greater than 0"))
 		}
+	case utilfeature.DefaultFeatureGate.Enabled(features.SplitStdoutAndStderr) && opts.Stream != nil:
```

Comment:
We do not include feature gates in validation. I would suggest moving the feature gate before you populate `PodLogOptions`.

Reply:
Got it